google/api/annotations.proto

```diff
@@ -102,3 +102,21 @@ extend google.protobuf.MethodOptions {
   // See `google/api/http.proto`.
   HttpRule http = 72295728;
 }
+
+
+// Common Resource Imports. (See `resources.proto` in this directory.)
+
+// A Resource representing a Project.
+// Useful to be able to declare a project as a referent for another field.
+option (google.api.resource_definition) = {
+  name: "Project"
+  path: "projects/{project}"
```

Comment:
In the design I think we are using `x_id` but I understand that this is probably has to `project` for back compat purposes. Thoughts?

s/{project}/{project_id}

Reply:
Yeah, I just grepped for the `organizations` resource path, and it's `organizations/{organization}` everywhere. It's also `projects/{project}` everywhere, except for within comments.

Reply:
Slight bummer. @chrisdunelm @lukesneeringer is my understanding that we should be using `x_id` by convention correct? If so lets add a comment saying this doesn't follow the convention for back-compat purposes.

Reply:
Done.

Reply:
In C# we auto-add the `id` suffix if it's missing. See here for an example in our generated `ProjectName` resource class: https://github.com/googleapis/gax-dotnet/blob/master/Google.Api.Gax/ResourceNames/ProjectName.cs#L67

This means that as far as C# is concerned `{project_id}` and `{organization_id}` are preferable, as we can then not auto-add an `id` suffix in the new generator.

@andreamlin How does Java handle this? (C# and Java are the only relevant languages, as I understand it)

Reply:
Java doesn't add an `_id` suffix, according to the philosophy of codegen that says that we take exactly what the config says and use that. Any non-essential preprocessing (essential preprocessing would be like escaping keywords on the surface) such as adding suffixes would have been done by configgen, or manual edits to the config.

Why do we have to have these be in the form `x_id`, other than convention? Whose convention is  it (Google [Cloud] API definition convention, C# surface for resource names convention, etc)?

Reply:
Convention is the only reason; there is no technical restriction on the name. The convention is from the one-platform team as I understand it; and it makes it clear that these are individual IDs that are the parts of resource-name.

Seeing that Java doesn't add the `_id` suffix, it looks like we'll need to keep the ids missing so as not to break Java compatibility. In C# we'll keep the suffix addition in the new micro-generator.

Reply:
So, to be clear, we want to use `_id` _nowhere_ in protos, correct? I assert the convention should be everywhere or nowhere.

Reply:
Alternatively we could use the `_id` suffix everywhere, with Java having a hardcoded (or yaml-based config) set of services where it knows to drop the suffix for backward compatibility.
I agree this isn't great, but would allow us to keep the suffix, which I understand is preferred?